build-aux/snap/snapcraft.yaml

```diff
@@ -127,3 +128,26 @@ parts:
       cp -a fc-cache-bionic $SNAPCRAFT_PART_INSTALL/bin/fc-cache-v7
     prime:
       - bin/fc-cache-v7
+  apparmor:
+    plugin: autotools
+    build-packages: [ bison, flex, gettext ]
+    source: https://launchpad.net/apparmor/3.0/3.0.1/+download/apparmor-3.0.1.tar.gz
+    override-build: |
+      cd libraries/libapparmor
+      ./autogen.sh
+      ./configure --prefix=/usr
+      make
+      # place libapparmor into staging area for use by snap-confine
+      make -C src install DESTDIR=$SNAPCRAFT_STAGE
+      cd ../../parser
+      make
+      mkdir -p $SNAPCRAFT_PART_INSTALL/usr/lib/snapd
+      cp -a apparmor_parser $SNAPCRAFT_PART_INSTALL/usr/lib/snapd/
```

Comment:
thinking about this some more, what runtime dependencies does apparmor_parser have? is it possible that when the snapd snap is installed on trusty for example and glibc and other things are older than the build-base of snapd snap (which is currently xenial but this may change in the future) that apparmor_parser fails from the snapd snap fails to run because we don't ship/use any runtime dependencies for apparmor in the snapd snap? 

I suppose we do ship libc for the non-static things in snapd like fc-cache, so perhaps it's just a matter of adjusting how apparmor_parser is called in the internal case to make sure that the dynamic library search path only includes things from the snapd snap, but I don't know if there are other dependencies that apparmor_parser needs which we would need to also ship with the snapd snap.

Reply:
@alexmurray could you comment on this?

Reply:
```
$ ldd /snap/snapd/current/usr/lib/snapd/apparmor_parser 
	linux-vdso.so.1 (0x00007ffd07170000)
	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007f8fcb24a000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f8fcb0fc000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f8fcaf10000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f8fcb27c000)
```

This is similar to snap-confine etc:

```
$ ldd /snap/snapd/current/usr/lib/snapd/snap-confine
	linux-vdso.so.1 (0x00007ffd4abde000)
	libudev.so.1 => /lib/x86_64-linux-gnu/libudev.so.1 (0x00007f38bfc3f000)
	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007f38bfc1d000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f38bfa31000)
	librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1 (0x00007f38bfa26000)
	libselinux.so.1 => /lib/x86_64-linux-gnu/libselinux.so.1 (0x00007f38bf9fa000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f38bfc78000)
	libpcre2-8.so.0 => /lib/x86_64-linux-gnu/libpcre2-8.so.0 (0x00007f38bf962000)
	libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007f38bf959000)
```